llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -962,22 +962,35 @@ void VPWidenIntrinsicRecipe::execute(VPTransformState &State) {
 
   // Use vector version of the intrinsic.
   Module *M = State.Builder.GetInsertBlock()->getModule();
+  bool IsVPIntrinsic = VPIntrinsic::isVPIntrinsic(VectorIntrinsicID);
   Function *VectorF =
-      Intrinsic::getOrInsertDeclaration(M, VectorIntrinsicID, TysForDecl);
-  assert(VectorF && "Can't retrieve vector intrinsic.");
+      IsVPIntrinsic
+          ? VPIntrinsic::getOrInsertDeclarationForParams(M, VectorIntrinsicID,
+                                                         TysForDecl[0], Args)
+          : Intrinsic::getOrInsertDeclaration(M, VectorIntrinsicID, TysForDecl);
+  assert(VectorF &&
+         "Can't retrieve vector intrinsic or vector-predication intrinsics.");
 
-  auto *CI = cast_or_null<CallInst>(getUnderlyingValue());
   SmallVector<OperandBundleDef, 1> OpBundles;
-  if (CI)
-    CI->getOperandBundlesAsDefs(OpBundles);
+  if (!IsVPIntrinsic) {
+    if (auto *CI = cast_or_null<CallInst>(getUnderlyingValue()))
+      CI->getOperandBundlesAsDefs(OpBundles);
+  }
 
-  CallInst *V = State.Builder.CreateCall(VectorF, Args, OpBundles);
+  Instruction *V = State.Builder.CreateCall(VectorF, Args, OpBundles);
```

Comment:
Why is  this needed?

Reply:
The next patch someone suggested
![image](https://github.com/user-attachments/assets/da8d003a-b4f3-439f-945f-b385d14fce10)